tez-runtime-library/src/main/java/org/apache/tez/runtime/library/common/shuffle/orderedgrouped/ShuffleScheduler.java

```diff
@@ -842,7 +846,7 @@ private void penalizeHost(MapHost host, int failures) {
     penalties.add(new Penalty(host, penaltyDelay));
   }
 
-  private int getFailureCount(InputAttemptIdentifier srcAttempt) {
+  private synchronized int getFailureCount(InputAttemptIdentifier srcAttempt) {
```

Comment:
this method is used only in a call path: isShuffleHealthy ->  isAbortLimitExceeedFor -> getFailureCount, which is not synchronized at all at the moment, could you help me understand how is making getFailureCount synchronized related to the actual patch?